crates/sui-types/src/object.rs

```diff
@@ -75,24 +82,29 @@ impl MoveObject {
         has_public_transfer: bool,
         version: SequenceNumber,
         contents: Vec<u8>,
-    ) -> Self {
+    ) -> Result<Self, SuiError> {
         // coins should always have public transfer, as they always should have store.
         // Thus, type_ == GasCoin::type_() ==> has_public_transfer
         debug_assert!(type_ != GasCoin::type_() || has_public_transfer);
-        Self {
+        if contents.len() > MAX_MOVE_OBJECT_SIZE {
+            return Err(SuiError::ObjectTooBig);
+        }
+        Ok(Self {
             type_,
             has_public_transfer,
             version,
             contents,
-        }
+        })
     }
 
     pub fn new_gas_coin(version: SequenceNumber, contents: Vec<u8>) -> Self {
-        unsafe { Self::new_from_execution(GasCoin::type_(), true, version, contents) }
+        // TODO: pass in value instead of contents to avoid unwrap
+        unsafe { Self::new_from_execution(GasCoin::type_(), true, version, contents).unwrap() }
     }
 
     pub fn new_coin(coin_type: StructTag, version: SequenceNumber, contents: Vec<u8>) -> Self {
-        unsafe { Self::new_from_execution(coin_type, true, version, contents) }
+        // TODO: pass in value instead of contents to avoid unwrap
```

Comment:
can we add comments why unwrap is safe here?

Reply:
Doing one better and changing the API to eliminate the `unwrap`